compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java

```diff
@@ -172,11 +172,13 @@ public static boolean isBallerinaProject(Path sourceRoot) {
      * @return organization name
      */
     public static String guessOrgName() {
-        String guessOrgName = System.getProperty(USER_NAME);
+        String guessOrgName = System.getProperty(USER_NAME).toLowerCase(Locale.getDefault());
         if (guessOrgName == null) {
             guessOrgName = "my_org";
         } else {
-            guessOrgName = guessOrgName.toLowerCase(Locale.getDefault());
+            if (!validateOrgName(guessOrgName)) {
+                guessOrgName =  guessOrgName.replaceAll("[^a-zA-Z0-9_]", "_");
```

Comment:
We should use the ballerina-toml schema for validation. 
https://github.com/ballerina-platform/ballerina-lang/blob/eb6b4cf9942c353267f862b36956b551d0bedc09/compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java#L145
We can check the exception message here and set alter org name instead of using the `ProjectUtils` util method

Reply:
This alteration is done only if the `Ballerina.toml` is not updated with package details. If it is already updated, then the usual diagnostic errors will be given asking the user to change it to something acceptable. This is the same for the project name as well. If it is not given, then we take the `user.name` system property by default, and that's when we handle special characters.

Please correct me if I'm wrong.

Reply:
oh ok, got it. We can go ahead with this change